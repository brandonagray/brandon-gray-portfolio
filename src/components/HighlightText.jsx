/*
  HIGHLIGHT_TEXT.JS
    Wrapper around text that draws an offset highlight on focus/hover
    or a custom trigger. Color can be configured. The toggle for drawing
    the highlight relies on its parent component to pass it the flag.
*/

import React from 'react';
import styled, { css } from 'styled-components';
import { mediaSize } from '../data/configOptions';

const Highlight = styled.span`
  position: relative;
  z-index: 110;
  width: 100%;

  &:before {
    /* background of title on hover */
    background-color: ${props => props.color};
    opacity: 0.6;
    content: '';
    position: absolute;
    top: 0.25em;
    left: 0.25em;
    right: -5px;
    height: 100%;
    width: 0;
    z-index: -1;

    transition: 250ms ease width;

    ${props =>
      props.hovered
        ? css`
            width: 95%;
            ${mediaSize.phone`
        width: 100%;
      `};
          `
        : null} ${mediaSize.phone`
      top: 0.4em;
      float: none;
      height: 0.9em;
      left: -3px;
      opacity: 0.75;
    `};
  }
`;

const HighlightText = props => (
  <Highlight color={props.color} hovered={props.hovered}>
    {props.children}
  </Highlight>
);

export default HighlightText;
