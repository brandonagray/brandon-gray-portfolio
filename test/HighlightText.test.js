import React from 'react';
import HighlightText from '../src/components/HighlightText';

describe('Highlight Text', () => {
  let text;

  describe('default state', () => {
    it('should be defined', () => {
      text = mount(
        <HighlightText color="#F8BJH3" hovered={false}>
          "Test text"
        </HighlightText>
      );
      expect(text).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = renderer
        .create(
          <HighlightText color="#F8BJH3" hovered={false}>
            "Test text"
          </HighlightText>
        )
        .toJSON();
      expect(tree).toHaveStyleRule('width', '0', {
        modifier: ':before'
      });
      expect(tree).toMatchSnapshot();
    });
  });

  describe('hovered state', () => {
    it('should expand bg on hover', () => {
      const tree = renderer
        .create(
          <HighlightText color="#F8BJH3" hovered={true}>
            "Test text"
          </HighlightText>
        )
        .toJSON();
      expect(tree).toHaveStyleRule('width', '95%', {
        modifier: ':before'
      });
      expect(tree).toMatchSnapshot();
    });
  });
});
