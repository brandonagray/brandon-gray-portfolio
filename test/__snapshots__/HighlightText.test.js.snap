// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight Text default state should render correctly 1`] = `
.c0 {
  position: relative;
  z-index: 110;
}

.c0:before {
  background-color: #F8BJH3;
  opacity: 0.6;
  content: '';
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: -5px;
  height: 1em;
  width: 0;
  z-index: -1;
  -webkit-transition: 250ms ease width;
  transition: 250ms ease width;
}

@media (max-width:26.5625em) {
  .c0:before {
    top: 0.4em;
    float: none;
    height: 0.9em;
    left: -3px;
    opacity: 0.75;
  }
}

<span
  className="c0"
  color="#F8BJH3"
>
  "Test text"
</span>
`;

exports[`Highlight Text hovered state should expand bg on hover 1`] = `
.c0 {
  position: relative;
  z-index: 110;
}

.c0:before {
  background-color: #F8BJH3;
  opacity: 0.6;
  content: '';
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: -5px;
  height: 1em;
  width: 0;
  z-index: -1;
  -webkit-transition: 250ms ease width;
  transition: 250ms ease width;
  width: 95%;
}

@media (max-width:26.5625em) {
  .c0:before {
    width: 100%;
  }
}

@media (max-width:26.5625em) {
  .c0:before {
    top: 0.4em;
    float: none;
    height: 0.9em;
    left: -3px;
    opacity: 0.75;
  }
}

<span
  className="c0"
  color="#F8BJH3"
>
  "Test text"
</span>
`;
